### PR TITLE
Modify PR template and document that tests are required

### PR DESCRIPTION
**Pull request by imanish003**
### Changes proposed in this Pull Request:

**Background:**
We discussed in Retrospective(pdnLyh-671-p2), that we should make it mandatory to include tests in PRs. The team agreed that moving forward, each PR should include test coverage as a standard requirement. Team members were encouraged to enforce this rigorously during reviews without guilt. This will help in maintaining the quality and reliability of the codebase. 

**Changes:**
- Updates the workflow to request reviewers to check for test coverage along with the testing instructions.

Closes # .

### How to test the changes in this Pull Request:

I believe testing is not required for this PR as it only changes text in workflow.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Ask reviewers to verify that relevant tests have been added or updated as needed.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by imanish003** [comment]
Hey @sunyatasattva 👋 

I have manually requested your review, as your input will be very valuable for this PR. Also, I am wondering if I should involve more folks from other teams to get their input on this PR because it is related to the whole WooCommerce monorepo. OR I can request in #woo-monorepo channel for review. Please let me know your thoughts on this :bow:

**Review by Aljullu — COMMENTED** [review]
Thanks for working on this, @imanish003!

> I have manually requested your review, as your input will be very valuable for this PR. Also, I am wondering if I should involve more folks from other teams to get their input on this PR because it is related to the whole WooCommerce monorepo. OR I can request in #woo-monorepo channel for review. Please let me know your thoughts on this 🙇

I was wondering the same, it would be good to get other teams feedback on this, given that the templates and workflows are shared.

Having said that, as a personal opinion, I'm not a big fan of introducing new checkboxes to the PR template. I feel it adds friction to the PR creation process and brings little benefits. As a middle point, we could mention it under _Submission Review Guidelines_, but without making it a checkbox that requires manual intervention. What do you think?

**Comment by imanish003** [comment]
Thanks @Aljullu for your feedback.

> I was wondering the same, it would be good to get other teams feedback on this, given that the templates and workflows are shared.

Makes sense. I will request reviews from other teams as well using the #woo-monorepo channel. Thanks for your input on this :bow:

> Having said that, as a personal opinion, I'm not a big fan of introducing new checkboxes to the PR template. I feel it adds friction to the PR creation process and brings little benefits. As a middle point, we could mention it under Submission Review Guidelines, but without making it a checkbox that requires manual intervention. What do you think?

I understand your concern. I also considered this while working on this PR. But it's easy to miss it in "Submission Review Guidelines" so I added it as a checkbox. But I agree that it might add friction to the PR creation process. We can definitely consider adding it under "Submission Review Guidelines" without making it a checkbox. This will also fulfill the purpose i.e.:
- We are encouraging contributors to add tests in PRs. 
- Along with this, we are encouraging reviewers to verify that relevant tests have been added or updated as needed. This way, reviewers can also remind contributors to add tests if they missed them without guilt.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (sunyatasattva) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] I have added/updated relevant tests (E2E, Unit, Integration etc.) to ensure my fix/feature works as intended, or this PR does not require tests.
```

**Review by sunyatasattva — APPROVED** [review]
I fixed a small typo.

I am ok with either way as I am not the main recipient of this change currently. Checkboxes may be overwhelming and I agree this change might be worth a check with all the other users of the monorepo.

With that said, I think we could also just proceed with the added comment notes. I think that's pretty uncontroversial. Thanks Manish for following up on this!

**Comment by imanish003** [comment]
> I think we could also just proceed with the added comment notes. I think that's pretty uncontroversial.

Thanks for your feedback @sunyatasattva. As adding a checkbox might be controversial, I also think it's better to proceed with the added comment notes. This also serves our intended purpose. I will proceed with this change. Thanks for your input on this :bow:

@Aljullu Please let me know if you have any other thoughts on this 🙂

**Comment by imanish003** [comment]
I have removed the changes from the PR template, therefore I don't think we need input from all the other users of the monorepo. I will wait for @Aljullu's approval and then I will merge this PR :bow:

Later, we can consider adding a checkbox in the PR template if needed.

**Review by Aljullu — APPROVED** [review]
LGTM, thanks for moving this forward, @imanish003!

**Pull request merged**
